app/components/main/view.jsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import TodoLayout from 'components/todo/layout';
+import Dashboard from 'components/dashboard/view';
+import session from 'services/session';
+
+export default class HeaderLayout extends React.Component {
+  state = session.store().getState()
+
+  constructor() {
+    super();
+
+    this.changeState = ::this.changeState;
```

Comment:
looks like a trick, could you clarify it for dummies a bit? what is a purpose of the binding above and how could it intersect with the `changeState` method defined below?